frontend/packages/ux-editor/src/components/config/componentSpecificContent/AttachmentList/AttachmentListInternalFormat.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { useState } from 'react';
+import { Fieldset, Switch } from '@digdir/design-system-react';
+import { AttachmentListContent } from './AttachmentListContent';
+import { useTranslation } from 'react-i18next';
+import { extractCurrentAvailableAttachments, isSelectionValid } from './attachmentListUtils';
+import { ArrayUtils } from '@studio/pure-functions';
+import type { AvailableAttachementLists, InternalDataTypesFormat } from './types';
+
+type AttachmentListInternalFormatProps = {
+  onChange: (selectedDataTypes: InternalDataTypesFormat) => void;
+  availableAttachments: AvailableAttachementLists;
+  internalDataFormat: InternalDataTypesFormat;
+};
+
+export const AttachmentListInternalFormat = ({
+  onChange,
+  availableAttachments,
+  internalDataFormat,
+}: AttachmentListInternalFormatProps) => {
+  const [dataTypesState, setDataTypesState] = useState<InternalDataTypesFormat>(internalDataFormat);
```

Comment:
I suggest adding a `useEffect` to update this state when the value of `internalDataFormat` changes (e.g. when adding or removing a component), something like :

```
  useEffect(() => {
    setDataTypesState(internalDataFormat);
  }, [internalDataFormat]);
```

Reply:
Thx! That will probably solve some of the issues you encountered :)

Reply:
I just noticed that Tomas had suggested it in a previous comment.
https://github.com/Altinn/altinn-studio/pull/12250#discussion_r1526252268

We also need to reset the value of `isValid` as he suggested:
```
  useEffect(() => {
    setDataTypesState(internalDataFormat);
    setIsValid(true);
  }, [internalDataFormat]);
```